sklearn/metrics/tests/test_classification.py

```diff
@@ -2905,3 +2905,241 @@ def test_brier_score_loss_deprecation_warning():
+    assert 0.5 < d2_score < 1.0
+
+    # check if poor predictions gives a relatively low value for the d2 score.
+    y_true = [0, 0, 0, 1, 1, 1]
+    y_pred = np.array(
+        [
+            [0.5, 0.5],
+            [0.1, 0.9],
+            [0.1, 0.9],
+            [0.9, 0.1],
+            [0.75, 0.25],
+            [0.1, 0.9],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    assert d2_score < 0
+
+    # check if simply using the average of the classes as the predictions
+    # gives a d2 score of 0
+    y_true = [0, 0, 0, 1, 1, 1]
+    y_pred = np.array(
+        [
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    assert d2_score == 0
+
+    # check if simply using the average of the classes as the predictions
+    # gives a d2 score of 0 when the positive class has a higher proportion.
+    y_true = [0, 1, 1, 1]
+    y_pred = np.array([[0.25, 0.75], [0.25, 0.75], [0.25, 0.75], [0.25, 0.75]])
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    assert d2_score == 0
+
+    # check if simply using the average of the classes as the predictions
+    # gives a d2 score of 0 where the y_true values are characters.
+    y_true = ["b", "c", "a", "d"]
+    y_pred = np.array(
+        [
+            [0.25, 0.25, 0.25, 0.25],
+            [0.25, 0.25, 0.25, 0.25],
+            [0.25, 0.25, 0.25, 0.25],
+            [0.25, 0.25, 0.25, 0.25],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    assert d2_score == 0
+
+    # check if simply using the average of the classes as the predictions
+    # gives a d2 score of 0 where the y_true values are characters and "a"
+    # has a relatively higher proportion.
+    y_true = ["b", "c", "a", "a"]
+    y_pred = np.array(
+        [
+            [0.5, 0.25, 0.25],
+            [0.5, 0.25, 0.25],
+            [0.5, 0.25, 0.25],
+            [0.5, 0.25, 0.25],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    assert d2_score == 0
+
+    # check if good predictions for character values gives a relatively
+    # higher d2 score.
+    y_true = ["b", "c", "a", "a"]
+    y_pred = np.array(
+        [
+            [0.1, 0.8, 0.1],
+            [0.1, 0.1, 0.8],
+            [0.8, 0.1, 0.1],
+            [0.8, 0.1, 0.1],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    assert 0.5 < d2_score < 1.0
+
+    # check if poor predictions for character values gives a relatively
+    # low value for the d2 score.
+    y_true = ["b", "c", "a", "a"]
+    y_pred = np.array(
+        [
+            [0.6, 0.3, 0.1],
+            [0.5, 0.2, 0.3],
+            [0.3, 0.4, 0.3],
+            [0.4, 0.5, 0.1],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    assert d2_score < 0
+
+    # check error if the number of classes are not equal.
```

Comment:
I would prefer to move all test checking for a "raise" into a separate (new) test.